impeller/core/resource_binder.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <optional>
+#include <string>
+
+#include "flutter/fml/logging.h"
+#include "flutter/fml/macros.h"
+#include "impeller/core/buffer_view.h"
+#include "impeller/core/formats.h"
+#include "impeller/core/sampler.h"
+#include "impeller/core/shader_types.h"
+#include "impeller/core/texture.h"
+
+namespace impeller {
+
+//------------------------------------------------------------------------------
+/// @brief      An interface for binding resources. This is implemented by
+///             |Command| and |ComputeCommand| to make CPU resources available
+///             to the GPU.
```

Comment:
It'd be more accurate to say it makes GPU resources (via handles) available to a given command's pipeline, since you can bind device private resources that never had a host buffer too (like we do with subpass render targets).

Reply:
Since this is causing friction for the Flutter-at-Google team I'm going to merge this and follow up with a doc fix.